TEST_APPS/icetea_plugins/nfc_test_parsers.py

```diff
@@ -0,0 +1,139 @@
+            try:
+                # iseeprom
+                value = PluginBase.find_one(line, "{{iseeprom=([\w]+)}}")
+                if value is not False:
+                    if ("TRUE" == value.upper() or "1" == value):
+                        results['iseeprom'] = True
+                    else:
+                        results['iseeprom'] = False
+                # max ndef
+                value = PluginBase.find_one(line, "{{maxndef=([0-9]+)}}")
+                if value is not False:
+                    results['maxndef'] = int(value)
+            except re.error as e: # the framework gobbles errors in the plugin
+                print("Regex error",e,"occured in",os.path.basename(__file__), "!!")
+                raise e
+        return results
+
+    def convert_from_hex(self, data):
+        value = []
+        for byte in data.split(' '):
+            if bool(byte):
+                if len(byte) > 2: # the line wrapping code for huge lines appears to munch the space character between 2 bytes
+                    value.append( int(byte[:2], 16))
+                    value.append( int(byte[2:4], 16))
+                else:
+                    value.append( int(byte, 16))
+        return value
+
+    # regex findall with additional filtering
+    @staticmethod
+    def find_all_hex_data(line, lookup):
+        """
+        regexp search with one value to return.
+
+        :param line: Line
+        :param lookup: regexp
+        :return: List of match groups or False
+        """
+        results = re.findall(lookup, line)
+        if results is not False:
+            results = [m for m in results if len(m) > 2] # remove empty matches and
+            # non hex-data result entries
+            return results
+        return False
+
+    def trace_parser(self, response):
+        results = {'iseeprom': None,        # 'true' if EEPROM
+                   'lastnfcerror':None,     # 0=OK >0 = error
+                   'nfcmessage':None,       # NDEF array of bytes
+                   'protocols':None,     # csv list
+                   'uri_id':None}        # nfc URI type identifier
+        respLines = response.lines
+        started_read_data = False
+        partial_data = ""
+        for line in respLines:
+            try:
+                value = PluginBase.find_one(line, "{{lastnfcerror=([0-9]+)}}")
+                if value is not False:
+                    results['lastnfcerror'] = int(value)
+                # {{nfcmessage=([0-9a-f\s]*)}}
+                data = PluginBase.find_one(line, "{{nfcmessage=([0-9a-f\s]*).*")
+                if data is not False:
+                    started_read_data = True
+                if started_read_data:  # read data until we see a }} pair
+                    values = self.find_all_hex_data(line, "([0-9a-f\s]*)")
+                    if values is not False:
+                        if "{{nfcmessage" in line:
+                            value = values[0]
```

Comment:
You have an if and then you ignore the result and use value[0] in both cases. I imagine you wanted to avoid including spurious results that include the matches from a line including nfcmessage?